aggregator/list.go

```diff
@@ -110,7 +112,7 @@ func newMetricList(resolution time.Duration, opts Options) *metricList {
 		resolution:    resolution,
 		flushInterval: flushInterval,
 		aggregations:  list.New(),
-		encoder:       msgpack.NewAggregatedEncoder(encoderPool.Get()),
+		encoder:       msgpack.NewAggregatedEncoder(encoder),
```

Comment:
It's somewhat conventional to do `object := pool.Get(); object.Reset()` before using the pooled object, this is also how we use pooled objects in m3db.